### PR TITLE
Add initial set of player tests

### PR DESCRIPTION
**Pull request by rolandkakonyi**
## Description

We have no meaningful tests for player behavior.

## Changes

Added 11 tests for basic player behavior testing.

## Checklist
- [x] 🗒 `CHANGELOG` entry - not applicable

### CODE REVIEW

File: integration_test/tests/exampleSpec.ts
Comment (rolandkakonyi) [inline integration_test/tests/exampleSpec.ts:1]: Removed this as we have actual tests now

File: integration_test/tests/playbackTest.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    spec.it('emits a Paused event after calling pause API', async () => {
+      await startPlayerTest({}, async () => {
+        await loadSourceConfig(Sources.artOfMotionHls);
+        await playFor(1);
+        await callPlayerAndExpectEvent((player) => {
+          player.pause();
+        }, EventType.Paused);
+      });
+    });
+  });
```
Comment (matamegger) [inline integration_test/tests/playbackTest.ts:37]: This should be two separate blocks I think.
```suggestion
  spec.describe('calling play when a source is loaded', () => {
    spec.it(
      'emits a Play and Playing events',
      async () => {
        await startPlayerTest({}, async () => {
          await loadSourceConfig(Sources.artOfMotionHls);
          await callPlayerAndExpectEvents((player) => {
            player.play();
          }, EventSequence(EventType.Play, EventType.Playing));
        });
      }
    );
  });
  spec.describe('calling pause when a source is loaded', () => {
    spec.it('emits a Paused event', async () => {
      await startPlayerTest({}, async () => {
        await loadSourceConfig(Sources.artOfMotionHls);
        await playFor(1);
        await callPlayerAndExpectEvent((player) => {
          player.pause();
        }, EventType.Paused);
      });
    });
  });
```

**Review by matamegger — APPROVED** [review]
Looks good. One small structural suggestion.

File: integration_test/tests/playbackTest.ts (hunk 1 above)
Comment (rolandkakonyi) [reply integration_test/tests/playbackTest.ts]: Applied feedback in 118b30c

**Pull request merged**